src/v/kafka/client/direct_consumer/fetcher.cc

```diff
@@ -361,6 +536,52 @@ fetcher::process_fetch_response(
         result.topics.push_back(std::move(topic_data));
     }
 
+    // Clear incremental fetch state, skipping partitions that errored
+    // or just returned new data.
+    for (const auto& to_process : partitions) {
+        const auto& included = to_process.to_include_in_fetch;
+        const auto& forgotten = to_process.to_forget;
+        const auto& topic = to_process.topic;
+
+        vassert(
+          included.empty() || forgotten.empty(), "Unexpectedly found both");
```

Comment:
please add some context into this vassert

Reply:
sure thing